### DISCUSSION TITLE
Error when trying to create wallet.

### DISCUSSION THREAD

**Discussion by joe86565** [cryptoadvance/specter-desktop#937] [Hardware Wallets]
Hi,

I get the error: "no keys selected for device. please try adding keys first".

Using coldcard and specter 1.0.0 on windows.

Have tried exporting to MicroSD as "Electrum Wallet" and "Generic JSON".

Any help appreciated.

Thanks in advance.

**Comment by ben-kaufman** [comment]
Hi, at which point do you get this message exactly?

I've made this tutorial for using ColdCard with Specter which might help https://twitter.com/_benkaufman/status/1333759877206904832?s=21

**Reply by joe86565** [reply]
Hi Ben,

Can you help with this?

**Reply by joe86565** [reply]
Hi Ben,

I created a wallet and received BTC.

The wallet was missing when I opened Specter today.

Can you point me to instructions on how to restore the wallet from backup PDF?

Thanks.

**Reply by ben-kaufman** [reply]
Hi, you can just click add wallet on the left sidebar, there you'll have an import wallet button where you can paste backup or scan the QR.

I'll check about the missing wallet issue, can you try maybe restart specter and recheck?

**Reply by joe86565** [reply]
restart did not work.

i tried to open specter when bitcoin core was not fully loaded and that may have caused the problem.

should i select the pdf backup if i select "choose file"?

**Reply by joe86565** [reply]
selecting pdf did not work.

all i have is coldcard_wallet_backup.pdf and addresses.csv.

i don't have scanning capabilities on this machine.

which specific data do i need to paste (from pdf)?

**Reply by ben-kaufman** [reply]
You shouldn't select the pdf itself, but you can copy paste the Advanced recovery details, "Specter Desktop Raw Data" section

**Reply by joe86565** [reply]
that seemed to work but no transactions yet. it is still re-scanning the blockchain.

i'll let you know when finished...

**Reply by joe86565** [reply]
what is the "verify on device" button for?

**Reply by joe86565** [reply]
also, should recent transactions only appear in specter or also in bitcoin core?

**Reply by ben-kaufman** [reply]
Transactions should be visible on both Specter and Core.

Verify on device is to verify the address on the device - to make sure the computer isn't lying to you and the address is legit.
Here I wrote some more info https://twitter.com/_benkaufman/status/1341700878072504321?s=21

**Reply by joe86565** [reply]
bitcoin core shows a 0 balance.

status from send end is "confirmed".

is bitcoin core waiting for specter to update?

**Reply by joe86565** [reply]
also, how can "verify on device" work with coldcard without connecting? there is no option for flash drive.

**Reply by ben-kaufman** [reply]
In bitcoin core did you select the wallet? should be at the top right corner.

For airgapped address verification coldcard has an address explorer which you can find on the main menu if you have the latest firmware, so you can use that

**Reply by joe86565** [reply]
no, i did not see / select wallet in the corner and must wait for the scan to complete before i can now.

yes, i verified with address explorer before making transactions. maybe a note about airgapped devices associated with "verify on device" would be helpful.

**Reply by joe86565** [reply]
scanning complete, balance is restored.

thank you!

have a good weekend.

**Comment by joe86565** [comment]
![Step #5](https://user-images.githubusercontent.com/78706907/107272743-c52e4980-6a13-11eb-8804-5451c5cb2a5a.PNG)
![Error (if I select add wallet before add device)](https://user-images.githubusercontent.com/78706907/107270934-27d21600-6a11-11eb-89b5-9f8ac0754c31.PNG)
![Error](https://user-images.githubusercontent.com/78706907/107270936-286aac80-6a11-11eb-93b1-58b06dff903a.PNG)
![Step #8](https://user-images.githubusercontent.com/78706907/107270937-286aac80-6a11-11eb-8a95-e4ce1e5dcd12.PNG)
![Step #9](https://user-images.githubusercontent.com/78706907/107270938-286aac80-6a11-11eb-9e45-149d2f27aea1.PNG)



Thanks for your help.

The error appears at step #10 after selecting "create wallet".

I notice these differences when following your steps exactly.

Step #5:
All XPUB data appears as "Custom" in mine.
Your example shows XPUB data in #0 Single Sig (nested), #0 Single Sig (segwit) and Custom.

Step #8:
Coldcard icon is grayed out and I can not select it as an option.

When I select continue...

Step #9
Your example shows Signer-1 Coldcard + Network, Purpous, etc. and mine does not.

I think the problem may have to do with XPUB data appearing in custom and not #0 Single Sig (nested) and #0 Single Sig (segwit).

Any ideas?

Thanks.



‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by joe86565** [comment]
Thanks for your help,

I changed the names in the json file and imported but it did not help.

I also changed the deriv & xfp to match the data in the example and it still won't create the wallet: "ERROR: No keys were selected for the device, please try adding keys first"

So it creates the device but won't read the keys for some reason.

I will try Telegram.

Thanks!



‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Reply by ben-kaufman** [reply]
Hi, sorry for the late response. Yes sounds like if it says all keys are "Custom" there might be a problem with the export. Which Coldcard firmware are you on?

**Comment by joe86565** [comment]
Hi Ben,

It's version 3.2.2

I sent the json to the coinkite team to make sure it was not corrupted during export.

This is their response:

"The file looks 100% okay.

One thought: in newer firmware versions, we renamed "p2sh-p2wpkh" from "p2wpkh-p2sh" which was non-standard. Perhaps Specter is sensitive to that change? This would only be an issue with BIP84 style wallets, and only the "name" field is changed."

I updated the names in the json as a test and it still did not create a wallet.

Please advise.

Thanks.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Reply by ben-kaufman** [reply]
Maybe you could share the file with me in DMs? (@_benkaufman of twitter or @benkaufman on telegram)

I just tested with v3.2.2 cold card and that looks fine...

**Comment by joe86565** [comment]
someone noticed that bitcoin core is in test mode. do you know how to switch out of test mode (i don't see anything in settings). i'll try this and then send the file if it is not the problem.

**Reply by ben-kaufman** [reply]
Oh you're connected to a testnet node? Then you'll not be able to create wallets with mainnet keys... that makes sense then.

If you want to use testnet really then you need to change your cold card to testnet mode...
If you want mainnet it depends how are you running Bitcoin Core exactly?

**Comment by joe86565** [comment]
i did not mean to be connected to a test node intentionally.

sorry, i don't understand your question about how i am running bitcoin core. what information do you need?

**Reply by ben-kaufman** [reply]
Where is Bitcoin Core running? You connected Specter to a node right? So how did you set up that node?
(btw the top left Bitcoin symbol is green right?)

**Comment by joe86565** [comment]
yes, top left symbol is green.

i thought i setup bitcoin core with default setting.

specter is set to local server.

**Reply by joe86565** [reply]
bitcoin jason-rpc config is set to auto detect

**Reply by ben-kaufman** [reply]
ok. In your bitcoin.conf file maybe you have testnet=1 ?

Did you run Bitcoin Core with the GUI app or via CLI?

**Reply by joe86565** [reply]
![network](https://user-images.githubusercontent.com/78706907/107520768-6386f000-6b77-11eb-8669-da9bc8bbc8f0.PNG)
![bitcoin core](https://user-images.githubusercontent.com/78706907/107520776-6681e080-6b77-11eb-819f-9dc83390962f.PNG)
![config](https://user-images.githubusercontent.com/78706907/107520789-6a156780-6b77-11eb-964f-82274a9f3331.PNG)

**Reply by joe86565** [reply]
i used: bitcoin-0.21.0-win64-setup.exe

**Reply by joe86565** [reply]
nothing in config about "test"

**Reply by joe86565** [reply]
![node window](https://user-images.githubusercontent.com/78706907/107521702-7d750280-6b78-11eb-9673-c3ffce88dc5c.PNG)

**Reply by ben-kaufman** [reply]
Very strange as it should default to mainnet... 
Maybe try to add testnet=0 in your bitcoin.conf file and restart it?

**Reply by joe86565** [reply]
ok, will try...

**Reply by joe86565** [reply]
still in test mode after adding testnet=0 to config and restart

**Reply by joe86565** [reply]
![start screen](https://user-images.githubusercontent.com/78706907/107522368-43583080-6b79-11eb-954d-5eddb4972160.PNG)

**Reply by ben-kaufman** [reply]
Starnge, where did you download bitcoin core from? Maybe try to reinstall? https://bitcoincore.org/en/download/

**Reply by joe86565** [reply]
i found the problem. i generated a shortcut to testnet instead of non-testnet to open the program. i have to wait for sync to complete because wallet is turned off...

**Reply by joe86565** [reply]
will let you know...

thanks!

**Reply by joe86565** [reply]
Hi Ben,

Works now, thanks for your help!
